primitives/native-token-management/src/lib.rs

```diff
@@ -86,26 +86,57 @@ mod inherent_provider {
 	use super::*;
 	use main_chain_follower_api::{DataSourceError, NativeTokenManagementDataSource};
 	use sidechain_mc_hash::get_mc_hash_for_block;
-	use sp_api::{ApiError, ProvideRuntimeApi};
+	use sp_api::{ApiError, Core, ProvideRuntimeApi};
 	use sp_blockchain::HeaderBackend;
+	use sp_version::RuntimeVersion;
 	use std::error::Error;
 	use std::sync::Arc;
 
 	pub struct NativeTokenManagementInherentDataProvider {
-		pub token_amount: NativeTokenAmount,
+		pub token_amount: Option<NativeTokenAmount>,
```

Comment:
It is, because if the version is too low, we return a provider containing `None`. `Some(0)` is possible because the type doesn't prevent that, it's on us to make sure it doesn't happen

Reply:
Just to clarify, in an earlier version this provider would not exist yet, and querying it would therefore return `None` (like you said) so we need to be backwards compatible with that?

Reply:
Yes, exactly. We need a no-op producer in the node, for blocks in which the pallet is absent (not yet added or already removed - this is a transitionary feature)